onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -0,0 +1,695 @@
+
+        // For the number of total elements in the input and output shapes
+        m_outputDataSize = std::accumulate(m_outputDims.begin(), m_outputDims.end(), 1, std::multiplies<uint32_t>());
+        m_inputDataSize = std::accumulate(m_inputDims.begin(), m_inputDims.end(), 1, std::multiplies<uint32_t>());
+
+        // { before_dft_axis, axis, after_dft_axis, real_or_complex }
+        std::array<uint32_t, 4> reshapedInputSize = { 1, 1, 1, m_inputDims.back() };
+        std::array<uint32_t, 4> reshapedOutputSize = { 1, 1, 1, m_outputDims.back() };
+
+        size_t reshapedIndex = 0;
+        for (int i = 0; i < m_inputDims.size() - 1; i++) {
+            if (i == m_axis || i == (m_axis + 1)) {
+                reshapedIndex++;
+            }
+            reshapedInputSize[reshapedIndex] *= m_inputDims[i];
+            reshapedOutputSize[reshapedIndex] *= m_outputDims[i];
+        }
+
+        auto temporarySize = reshapedInputSize;
+        temporarySize.back() = reshapedOutputSize.back();
+
+        // Calculate elements and strides
+        std::array<uint32_t, 4> reshapedInputStrides = { 1, 1, 1, 1 };
+        std::array<uint32_t, 4> reshapedOutputStrides = { 1, 1, 1, 1 };
+        std::array<uint32_t, 4> temporaryStrides = { 1, 1, 1, 1 };
+        for (int i = static_cast<int>(m_inputDims.size()) - 2; i >= 0; i--) {
+            reshapedInputStrides[i] = reshapedInputSize[i + 1] * reshapedInputStrides[i + 1];
+            reshapedOutputStrides[i] = reshapedOutputSize[i + 1] * reshapedOutputStrides[i + 1];
+            temporaryStrides[i] = temporarySize[i + 1] * temporaryStrides[i + 1];
+        }
+
+        // Get DFT Length
+        auto dftLength = m_inputDims[m_axis];
+
+        // Calculate passes
+        m_numPasses = static_cast<unsigned>(log2(dftLength));
+        bool hasOnePass = m_numPasses == 1;
+        bool hasOddPasses = m_numPasses % 2;
+        bool hasEvenPasses = !hasOddPasses;
+
+        // write directly input buffer to output buffer, dont create temps
+        bool writeToOutput = hasOnePass;
+        // First and final are input/output buffers, but all else ocillate between 2 temp buffers
+        bool occilateBetweenTwoTemporaries = !hasOnePass && m_isOnesided;
+        // First is input buffer, all else ocillate between temp and output, causing the final pass to write to the output buffer
+        bool occilateFirstOutputThenTemporary = hasOddPasses && !m_isOnesided;
+        // First is input buffer, all else ocillate between output and temp, causing the final pass to write to the output buffer
+        bool occilateFirstTemporaryThenOutput = hasEvenPasses && !m_isOnesided;
+
+        // Create the resource loop list
+        // Add the input resource to the loop list
+        resourceLoopList_.push_back({});
+        resourceLoopList_.back().Resource = nullptr;
+        resourceLoopList_.back().Sizes = reshapedInputSize;
+        resourceLoopList_.back().Strides = reshapedInputStrides;
+
```

Comment:
It will be nice when ORT updates to C++20 so we can just say

```
        resourceLoopList_.push_back({
            .Resource = nullptr;
            .Sizes = reshapedInputSize;
            .Strides = reshapedInputStrides;
        });
```

Reply:
yeah!